setuptools/dist.py

```diff
@@ -402,6 +405,23 @@ def _normalize_requires(self):
             (k, list(map(str, _reqs.parse(v or [])))) for k, v in extras_require.items()
         )
 
+    def _finalize_license_expression(self) -> None:
+        """Normalize license and license_expression."""
+        license_expr = self.metadata.license_expression
+        if license_expr:
+            normalized = canonicalize_license_expression(license_expr)
+            if license_expr != normalized:
+                InformationOnly.emit(f"Normalizing '{license_expr}' to '{normalized}'")
+                self.metadata.license_expression = normalized
+
+            for cl in self.metadata.get_classifiers():
+                if not cl.startswith("License :: "):
+                    continue
+                raise InvalidConfigError(
+                    "License classifier are deprecated in favor of the license expression. "
+                    f"Remove the '{cl}' classifier."
+                )
```

Comment:
Because it is a deprecation, should we give a `warning` first before evolving to an error? (we can skip the deprecated classifier in the meantime).

Reply:
I'd be open to that as well. My reasoning for the error was that the users isareupdating the metadata anyway. Might as well error then. Deprecations tend to get unnoticed unfortunately, especially with CD pipelines.